### ISSUE TITLE
dnaPipeTE vs RepeatModeler

### ISSUE THREAD

**Issue by r-mashoodh** [clemgoub/dnaPipeTE#59]
Hi Clement, 

Thanks for developing this neat tool. This is more of a question than issue. When I ran RepeatModeler2 on my species genome I get about 2/3 of consensi are classified as 'unknown' and I am struggling to classify these. 

`RepeatModeler -database genome -engine ncbi -pa 32 -LTRStruct`

Interestingly, dnaPipeTE (with some whole genome sequencing data) is much better at classifying and only comes up with about 1/3 of the TEs as being 'unknown'. 

Why is this? Is it to do with a non-model contiguous genome or something else? 

cheers!

**Comment by r-mashoodh** [comment]
ps. Strangely ... none of the RepeatModeler2 consensi were classified as simple/low-complexity either ...

**Comment by clemgoub** [comment]
Dear @r-mashoodh,

Thank you for your message! 

The main difference is in the homology methods and thresholds used to classify the TE. In my experience, RepeatModeler (through its package RepeatClassifier) is more conservative. It uses nucleotides and protein homology to classify the consensus it creates.

**Comment by r-mashoodh** [comment]
Dear Clément,

Thanks for the info! Super helpful as usual. What species have you used DeepTE and ClassifyTE with? I'm curious how much your results diverged from RepeatModeler. I'm working with insects and I'm a bit thrown off by how much they disagree with things classified in RepeatModeler2!

cheers
R

**Comment by clemgoub** [comment]
Yes, I agree with you. I only tested DeepTE on the rice weevil, but it was
not agreeing in many cases with RM2 (RepeatClassifier). We ended up
combining different methods to resolve the classification. I would be
interested to look further into ClassfyTE as well but did not get a chance.

I think that one limit of these machine learning methods is that it focuses
on the k-mer frequency. It is often enough to solve the problem but within
the limit of your train set and what is considered "true TE set" and
"not-TE set" for the training. Including methods based on the structure
(for example reporting LTR or TIR) can be a great added info as well as
using the relative position of different features in a TE consensus. I am
sending you the link to our preprint where we explore one way to build a TE
library. There will be a few minor changes in the final publication, but
the TE classification method can be interesting for you.
https://www.biorxiv.org/content/10.1101/2021.03.03.408021v4

Cheers,

Clément
